vent/io/devices/pins.py

```diff
@@ -177,21 +175,15 @@ def __hardware_pwm(self, frequency, duty):
 
         Args:
             frequency: A new PWM frequency to use.
-            duty (int): A pigpio integer representation of duty cycle
+            duty (int): The PWM duty cycle to set. Must be between 0 and 1.
         """
-        # print('pin: %3.0d freq: %5.0d duty: %4.2f'%(self.pin,frequency,duty))
-        result = self.pig.hardware_PWM(self.pin, frequency, duty)
-        if result != 0:
-            self.hardware_enabled = False
-            raise RuntimeWarning(
-                'Failed to start hardware PWM with frequency {} on pin {}. Error: {}'.format(
-                    frequency,
-                    self.pin,
-                    error_text(result)
-                )
-            )
-        else:
-            self.hardware_enabled = True
+        try:
+            self.pig.hardware_PWM(self.pin, frequency, duty)
+            self._hardware_enabled = True
+        except Exception:

```

Comment:
what is the exception that we're filtering here? this seems like a p broad catch w/o comment

Reply:
yep good catch - should be pigpio.error